### PR TITLE
GNUmakefile: add "help" target

### PR DESCRIPTION
**Pull request by dkegel-fastly**
Example output:

```
$ make help
clean                           Remove build directory
fmt                             Reformat source
fmt-check                       Warn if any source needs reformatting
gen-device                      Generate microcontroller-specific sources
llvm-source                     Get LLVM sources
llvm-build                      Build LLVM
lint                            Lint source tree
spell                           Spellcheck source tree
```

Might even work on windows, since git for windows comes with awk.

### CODE REVIEW

**Review by archie2x — APPROVED** [review]
Nice.

File: GNUmakefile [hunk 1]
```diff
@@ -291,8 +289,7 @@ ifeq (, $(shell which node))
 endif
 	@if [ $(NODEJS_VERSION) -lt $(MIN_NODEJS_VERSION) ]; then echo "Install NodeJS version 18+ to run tests."; exit 1; fi
 
-# Build the Go compiler.
-tinygo:
+tinygo: ## Build the Go compiler
```
Comment (leongross) [inline GNUmakefile]: nit: Maybe make it 
```suggestion
tinygo: ## Build the TinyGo compiler
```
to make a distinction from big go?

**Review by leongross — CHANGES_REQUESTED** [review]

**Review by archie2x — COMMENTED** [review]
Mention this in BUILDING.md ?

```patch
diff --git a/BUILDING.md b/BUILDING.md
index dfc069c0..c26650fd 100644
--- a/BUILDING.md
+++ b/BUILDING.md
@@ -28,6 +28,21 @@ build tools to be built. Go is of course necessary to build TinyGo itself.
 The rest of this guide assumes you're running Linux, but it should be equivalent
 on a different system like Mac.
 
+## Help
+
+The GNUmakefile has a 'help' target
+
+    % make help
+    clean                           Remove build directory
+    fmt                             Reformat source
+    fmt-check                       Warn if any source needs reformatting
+    gen-device                      Generate microcontroller-specific sources
+    llvm-source                     Get LLVM sources
+    llvm-build                      Build LLVM
+    tinygo                          Build the Go compiler
+    lint                            Lint source tree
+    spell                           Spellcheck source tree
+
 ## Download the source
 
 The first step is to download the TinyGo sources (use `--recursive` if you clone
```

File: GNUmakefile (hunk 1 above)
Comment (dkegel-fastly) [reply GNUmakefile]: Done

**Comment by dkegel-fastly** [comment]
> Mention this in BUILDING.md ?

Done.

**Review by ydnar — APPROVED** [review]
LGTM. Thank you Dan!

**Comment by deadprogram** [comment]
@dkegel-fastly do you want to squash these commits before merge? or do you want me to do so for you?

**Comment by dkegel-fastly** [comment]
I'll squash and merge now that you're had a chance to look, thanks.

**Pull request merged**